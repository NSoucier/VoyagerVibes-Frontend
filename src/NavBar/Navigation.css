#toggler {
    border: none;
    color: rgba(0, 0, 0, 0);
}

#links {
    text-decoration: none;
    color: black;
}

#nav {
    background-color: rgba(234, 212, 255, 0.933);
    border-bottom: 3px solid rgba(156, 69, 156, 0.422);
}

.navbar-brand {
    font-family: "Pacifico", cursive;
    font-weight: 400;
    font-style: normal;
}