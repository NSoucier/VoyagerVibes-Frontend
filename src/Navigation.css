#toggler {
    border: none;
    color: rgba(0, 0, 0, 0);
}

#links {
    text-decoration: none;
    color: black;
}

#nav {
    background-color: rgba(230, 230, 250, 0.933);
    border-bottom: 3px solid rgba(128, 0, 128, 0.636);
}

.navbar-brand {
    font-family: "Pacifico", cursive;
    font-weight: 400;
    font-style: normal;
}