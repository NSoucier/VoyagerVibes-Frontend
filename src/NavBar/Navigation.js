import React, { useState } from "react";
import { Link } from "react-router-dom";
import { Collapse, Navbar, NavbarToggler, NavbarBrand, Nav, NavItem, NavLink } from 'reactstrap';
import './Navigation.css';
  
  function Navigation({ logout }) {
    const [openNav, setOpenNav] = useState(false);
    const toggle = () => setOpenNav(!openNav);

    // logs out user
    function logoutUser() {
      logout();
    }
  
    return (
      <div>
        <Navbar expand="md" fixed="top" id="nav">
          <NavbarBrand><Link id="links" to='/'>VoyagerVibes</Link></NavbarBrand>
          <NavbarToggler onClick={toggle} id="toggler"/>
          <Collapse isOpen={openNav} navbar>
            { localStorage.user ? (
              <Nav className="ml-auto" navbar>
                <NavItem>
                  <NavLink><Link id="links" to='/explore'>Explore</Link></NavLink>
                </NavItem>
              </Nav>              
              ) : ''
            }
            <Nav className="ms-auto" navbar>
              { localStorage.user ? 
                (
                  <>
                    <NavItem >
                      <NavLink><Link id="links" to='/profile'>{localStorage.user}</Link></NavLink>
                    </NavItem>
                    <NavItem>
                      <NavLink><Link id="links" to='/' onClick={logoutUser}>Logout</Link></NavLink>
                    </NavItem>                
                  </>
                )
                : (
                  <>
                    <NavItem >
                      <NavLink><Link id="links" to='/login'>Login</Link></NavLink>
                    </NavItem>
                    <NavItem>
                      <NavLink><Link id="links" to='/signup'>Sign Up</Link></NavLink>
                    </NavItem>                
                  </>
                )
              }
            </Nav>
          </Collapse>
        </Navbar>
      </div>
    );
  }
  
  export default Navigation;