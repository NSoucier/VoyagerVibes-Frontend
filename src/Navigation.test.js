import React, { act } from "react";
import { render, fireEvent } from "@testing-library/react";
import Navigation from "./Navigation";
import { Route, Routes, MemoryRouter, BrowserRouter } from "react-router-dom";

// smoke test
it("renders without crashing", async function () {
    await act(() => {
        render(
            <BrowserRouter>
                <Navigation />
            </BrowserRouter>
          );
    });
});

// snapshot test
it("matches snapshot", function() {
    const {asFragment} = render(
        <BrowserRouter>
            <Navigation />
        </BrowserRouter>
      );
    expect(asFragment()).toMatchSnapshot();
});


