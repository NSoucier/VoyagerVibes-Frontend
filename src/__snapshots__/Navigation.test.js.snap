// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md fixed-top"
      id="nav"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
        />
        <a
          href="/"
          id="links"
        >
          VoyagerVibes
        </a>
        <button
          aria-label="Toggle navigation"
          class="navbar-toggler"
          id="toggler"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
        >
          <ul
            class="ms-auto navbar-nav"
          >
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
              />
              <a
                href="/login"
                id="links"
              >
                Login
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
              />
              <a
                href="/signup"
                id="links"
              >
                Sign Up
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
